### PR TITLE
test(e2e): reduce pool capitalization flakiness

### PR DESCRIPTION
**Pull request by p0mvn**
Reduces pool capitalization flakiness by:
- Increasing min liquidity cap threshold to 100K
- Adding alloyed pool IDs to exclusion since broken on Numia side

## Summary by CodeRabbit

- **New Features**
	- Implemented a caching mechanism for token price retrieval, improving response times and reducing API calls.
  
- **Bug Fixes**
	- Adjusted the minimum liquidity capitalization threshold in tests to enhance reliability.
	- Updated conditions for skipping tests based on pool liquidity to ensure only suitable pools are evaluated.

- **Chores**
	- Expanded the list of problematic pool IDs to exclude specific pools from testing.

### CODE REVIEW

File: tests/coingecko_service.py [hunk 1]
```diff
@@ -4,9 +4,13 @@
 USD_CURRENCY = "usd"
 
 class CoingeckoService:
+    cache = {}
```
Comment (p0mvn) [inline tests/coingecko_service.py:7]: Note: drive-by optimization

File: tests/test_pools.py [hunk 2]
```diff
@@ -50,32 +51,35 @@ def run_pool_liquidity_cap_test(environment_url, pool_data):
...
 
-        actual_error = relative_error(sqs_liquidity_cap, pool_liquidity)
+    pool_liquidity = pool_data.get("liquidity")
 
-        assert actual_error < error_tolerance, f"ID ({pool_id}) Pool liquidity cap was {sqs_liquidity_cap} - expected {pool_liquidity}, actual error {actual_error} error tolerance {error_tolerance}" 
-    else:
-        pytest.skip("Pool liquidity is too low - skipping to reduce flakiness")
+    sqs_pool = sqs_service.get_pools(pool_ids=pool_id)
+ 
+    sqs_liquidity_cap = int(sqs_pool[0].get("liquidity_cap"))
```
Comment (p0mvn) [inline tests/test_pools.py:78]: Note: drive by refactor for readability

File: tests/test_pools.py [hunk 3]
```diff
@@ -10,7 +11,7 @@
 
 # Min liquidity capitalization in USDC for a pool to be considered
 # in tests. Arbitrarily chosen as to avoid flakiness.
-min_pool_liquidity_cap_usdc = 50_000
+min_pool_liquidity_cap_usdc = 100_000
 num_all_pools_expected = 1500
```
Comment (p0mvn) [inline tests/test_pools.py:15]: Core change: 1.

Increasing the filter. We do not need to test this against every pool. As long as ~20 pools are correct, we can extrapolate with confidence

File: tests/test_pools.py [hunk 4]
```diff
@@ -50,32 +51,35 @@ def run_pool_liquidity_cap_test(environment_url, pool_data):
     # See: https://linear.app/osmosis/issue/NUMIA-35/missing-data-for-white-whale-pool
     skip_whitewhale_code_id = 641
     # This pool has a bug in the Numia side.
-    skip_alloyed_pool_id = [1816, 1878]
+    skip_alloyed_pool_id = [1816, 1868, 1878, 1925]
```
Comment (p0mvn) [inline tests/test_pools.py:54]: Added alloyed pools that are broken Numia-side

**Review by deividaspetraitis — APPROVED** [review]
Looks good to me! :+1:

**Comment by p0mvn** [comment]
i was wondering why the tests are still broken. Apparently, I never merged the fix PR 😆

**Pull request merged**